Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,184 @@
+
+            $p4 = $this->findFunction($tokens, $className, $classStart, $classEnd);
+
+            if (!$p4) {
+                // No PHP4-constructor!
+                continue;
+            }
+
+            $p5 = $this->findFunction($tokens, '__construct', $classStart, $classEnd);
+
+            if (!$p5) {
+                // No PHP5-constructor, we can rename the old one
+                $tokens[$p4['nameIdx']]->setContent('__construct');
+                continue;
+            }
+
+            // Does the PHP4-constructor call __construct?
+            if (false !== stripos($p4['body'], '$this->__construct')) {
+                // Good, delete it!
+                for ($i = $p4['startIdx']; $i <= $p4['endIdx']; ++$i) {
+                    $tokens[$i]->clear();
+                }
+            } elseif (false !== stripos($p5['body'], '$this->'.$className)) {
+                // It looks like __construct calls the PHP4 constructor:
+                // That's pretty bad, delete the new one
+                for ($i = $p5['startIdx']; $i <= $p5['endIdx']; ++$i) {
+                    $tokens[$i]->clear();
+                }
+                // And rename the old one
+                $tokens[$p4['nameIdx']]->setContent('__construct');
+            }
+        }
+
+        // Check calls to the parent constructor
+        foreach ($tokens->findGivenKind(T_EXTENDS) as $index => $token) {
+            $parIndex = $tokens->getNextMeaningfulToken($index);
+            $parentClass = $tokens[$parIndex]->getContent();
+
+            // using parent::ParentClassName()
+            $currIndex = $parIndex;
+            while ($currIndex = $tokens->getNextTokenOfKind($currIndex, array(array(T_PAAMAYIM_NEKUDOTAYIM, "::")))) {
+                $prevIndex = $tokens->getPrevMeaningfulToken($currIndex);
+                if ($tokens[$prevIndex]->equals(array(T_STRING, 'parent'))) {
+                    $currIndex = $tokens->getNextMeaningfulToken($currIndex);
+                    if (!strcasecmp($tokens[$currIndex]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$currIndex]->setContent('__construct');
+                        break;
+                    }
+                }
+            }
+
+            // using $this->ParentClassName()
+            $currIndex = $parIndex;
+            while ($currIndex = $tokens->getNextTokenOfKind($currIndex, array(array(T_OBJECT_OPERATOR, "->")))) {
+                $prevIndex = $tokens->getPrevMeaningfulToken($currIndex);
+                if ($tokens[$prevIndex]->equals(array(T_VARIABLE, '$this'))) {
+                    $operatorIndex = $currIndex;
+                    $currIndex = $tokens->getNextMeaningfulToken($currIndex);
+                    if (!strcasecmp($tokens[$currIndex]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$prevIndex]->setContent('parent');
```

Comment:
this seems wrong to me, because we will end up with a token ``array(T_VARIABLE, 'parent')``, and I doubt that ``parent`` is parsed as a ``T_VARIABLE`` token